### PR TITLE
array: implement missing routines

### PR DESCRIPTION
**Pull request by lolBig**
implemented the blow array prototypes
- Array.prototype.find
- Array.prototype.findIndex
- Array.prototype.fill
- Array.prototype.includes
- Array.prototype.copyWithin

### CODE REVIEW

File: deps/jerry/tests/jerry/array-prototype-includes.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+(function() {
+  console.log([].includes.call(arguments, 'a')); // true
+  console.log([].includes.call(arguments, 'd')); // false
+})('a','b','c');
+
+// 数组长度是3
```
Comment (algebrait) [inline deps/jerry/tests/jerry/array-prototype-includes.js]: Would you please use english?

**Comment by yorkie** [comment]
@lolBig Could we put the following in another patch, and let's land implemented asap:

- Array.from
- Array.of
- String.prototype.startsWith
- String.prototype.endsWith
- String.prototype.repeat
- Object.is

File: deps/jerry/tests/jerry/array-prototype-includes.js (hunk 1 above)
Comment (lolBig) [reply deps/jerry/tests/jerry/array-prototype-includes.js]: sorry, forget to remove hints fro myself, i will remove it later

**Comment by lolBig** [comment]
> @lolBig Could we put the following in another patch, and let's land implemented asap:
> 
> Array.from
> Array.of
> String.prototype.startsWith
> String.prototype.endsWith
> String.prototype.repeat
> Object.is

still missing unit tests, i will finished them in next week

**Comment by yorkie** [comment]
@lolBig This is not relevant with testing, I hope every PR will not be that large, which would hard to review, track and manage conflicts.

**Comment by lolBig** [comment]
@yorkie yes, this pr looks a bit large
please wait for me to finish the unit tests for the implemented methods

**Comment by yorkie** [comment]
Sure @lolBig :)

File: CMakeLists.txt [hunk 2]
```diff
@@ -20,6 +20,8 @@ set(IOTJS_VERSION_MAJOR 0)
 set(IOTJS_VERSION_MINOR 10)
 set(IOTJS_VERSION_PATCH 0)
 
+link_directories(/usr/local/lib)
```
Comment (yorkie) [inline CMakeLists.txt]: Can remove this conflicts line?

File: deps/jerry/jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c [hunk 3]
```diff
@@ -1963,6 +1963,7 @@ ecma_builtin_array_prototype_object_every (ecma_value_t this_arg, /**< this argu
           ret_value = ECMA_VALUE_FALSE;
         }
 
+        ecma_free_value(current_index);
```
Comment (yorkie) [inline deps/jerry/jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c]:
```suggestion
        ecma_free_value (current_index);
```

**Comment by yorkie** [comment]
@algebrait Could you please have a review again :)

**Pull request merged**